application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,213 @@
+        implements MessageContextCommand {
+    private static final Logger logger = LoggerFactory.getLogger(TransferQuestionCommand.class);
+    private static final String COMMAND_NAME = "transfer-question";
+    private static final String TRANSFER_QUESTION_TITLE_ID = "transferID";
+    private static final String TRANSFER_QUESTION_INPUT_ID = "tranferQuestion";
+    private static final String TRANSFER_QUESTION_TAG = "tags";
+    private static final String QUESTIONS_CHANNEL_ID = "1051826284008853505";
+    private String originalMessageId;
+    private String originalChannelId;
+    private static final int TITLE_MAX_LENGTH = 50;
+
+
+    private static final Pattern TITLE_COMPACT_REMOVAL_PATTERN = Pattern.compile("\\W");
+    static final int TITLE_COMPACT_LENGTH_MIN = 2;
+    static final int TITLE_COMPACT_LENGTH_MAX = 30;
+
+
+
+    /**
+     * Creates a new instance.
+     */
+    public TransferQuestionCommand() {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+    }
+
+    /**
+     * Verifies if user has role higher or equal to Community Ambassador in terms of hierarchy.
+     */
+    public boolean hasRequiredRoles(MessageContextInteractionEvent event) {
+
+        List<Role> userRoles = event.getMember().getRoles();
+        List<Role> communityAmbassador =
+                event.getGuild().getRolesByName("COMMUNITY AMBASSADOR", true);
+
+        for (Role role : userRoles) {
+            if (role.getPosition() >= communityAmbassador.get(0).getPosition()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+
+        if (!hasRequiredRoles(event)) {
+            event.reply("You're not authorised to use this command").setEphemeral(true).queue();
+        }
+
+        String originalMessage = event.getTarget().getContentRaw();
+
+
+        originalMessageId = event.getTarget().getId();
+        originalChannelId = event.getChannel().getId();
+
+        String author = event.getTarget().getAuthor().getId();
+
+
+        TextInput transferQuestionTitle =
+                TextInput.create(TRANSFER_QUESTION_TITLE_ID, "Title", TextInputStyle.SHORT)
+                    .setMaxLength(70)
+                    .setMinLength(8)
+                    .setValue(createTitle(originalMessage))
+                    .build();
+
+        TextInput transferQuestionInput = TextInput
+            .create(TRANSFER_QUESTION_INPUT_ID, "Transfer question menu", TextInputStyle.PARAGRAPH)
+            .setValue(originalMessage)
+            .setRequiredRange(3, 1000)
+            .build();
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
+            .build();
+
+
+
+        String transferQuestionModalComponentID = generateComponentId(author);
```

Comment:
also this makes sense, now the operations looks more generic with doSomethingWithMessage(args.get(i)) rather than doSomethingWithMessage(originalMessageId), which it's easy to tell what arguement is being passed on.